### ISSUE TITLE
Live Twitch Stream Recording Stops at random timings ()

### ISSUE THREAD

**Issue by Elandrya2711** [Zibbp/ganymede#447]
After Recording for a random time ( i had everything from 34 Minutes to little over 2 hours ) the recording stops and starts with the post processing. The Stream is still running at this time. Settings are all default, i did a fresh install using docker.

[download] Written 3.08 GiB to …9e3a-0242ac130006-video.mp4 (54m02s @ 0 bytes/s)
[download] Written 3.08 GiB to …9e3a-0242ac130006-video.mp4 (54m02s @ 0 bytes/s)
[download] Written 3.08 GiB to …9e3a-0242ac130006-video.mp4 (54m02s @ 0 bytes/s)
[download] Written 3.08 GiB to …9e3a-0242ac130006-video.mp4 (54m02s @ 0 bytes/s)
[stream.hls][warning] Failed to reload playlist: Unable to open URL: https://video-weaver.arn04.hls.ttvnw.net/v1/playlist/Cv0E6CxwrBvRZ5bf_-wQxLFFCLb2Z36WZl9PdKz_BI8GmazgxeyddNhRBeczKabOhaQ8Pjjxb_dQGtUtfNdf1180wS1pctZ-Fr4RnqSR-NIkRb8svL00GSydxPA2saPqnpcYRoyQKNmVh6Z98QLrhibDLLQNTEix2XO4U8BTTWDK-fmUyrNDbhSkFVUSj4BrUYDKtNCL35QiKmn-AD-KA0Hu6lgr1GOjQJUVjLoskISwwlMW1hoT9bXHHSZR4f-H4NbPiaRdLRE5Yaoq7xKbNZzMHn5lA7fldOTBRe-wO3Koz1nAsvnhYzC0PcNa-gB4jZUaNeR2yYtyVQoFT60aGQ4U1DlNMncbzu7gSkvxITjqJdAt0YhiTB1ddMmBgq4dtgf_Ey43FxS6EVf7xr-latb999i4nOCyThXI4AO1Vk_SxB8rQkdzAw7Wc1gzNCsRjLcylxvOCo5Z_gY2q3B51nmpO9E1n7-NISLI_r2NsfR9Ad79Rcv8e5Pbb1RHA_N3NTdr1-0EJiAw94y1t-WKuP7wpqPe0aTIBB3mpE_-dr62yYe_qYt9INQ1WEYLSjgmG7TkLYgjZn6_1H1LdWDh7xhaLYzJ-O5R4MzgcH28kbzerL30MwsHZN6ttAIO6_N0XQJl8AIxND0eV-b90junp3xG2ePm6GsGj7yk7lOpiW-F8lIEtrWwEFiW-ssRDZMXdai6k_lR0DcER1OG03wxcZurEEmzUJFASAegElWqgYgBHGWH6C6Qi6-UOVkwi2Plm2fRTN-cPutBsdqyx1exV2QmUBFmVtqK1lFw3eiMZwWbpmWQJt6tcCp1_4A29MdyDMvBp2lF54dg0TjpSGNc7xoM5ypinQKO1ySVab4SIAEqCWV1LXdlc3QtMjDRCQ.m3u8 (HTTPSConnectionPool(host='video-weaver.arn04.hls.ttvnw.net', port=443): Read timed out. (read timeout=20.0))
[cli][info] Stream ended
[cli][info] Closing currently open stream...
error: Error when reading from stream: Read timeout, exiting

Any idea how to debug/fix this? Thx for your help.

**Comment by Zibbp** [comment]
Couple of questions
- Do you have live stream proxying enabled in Admin > Settings?
- Is your internet cutting out when this happens?

**Comment by Elandrya2711** [comment]
I am not using any proxy. It's a fresh install.  It's running on an proxmox cluster with 2 redundant gateways. There shouldn't be any internet problems.

**Comment by Elandrya2711** [comment]
I just tried again:
[download] Written 541.25 MiB to …3a-0242ac130006-video.mp4 (12m38s @ 0 bytes/s)
[stream.hls][warning] Failed to reload playlist: Unable to open URL: https://video-weaver.arn04.hls.ttvnw.net/v1/playlist/Cu4EfeHqUkMOY8HJf-tzp0ErXfge3iUzlTrxSIj1sfHg1iOfbCLOk2wD3AV02Mnilht1jjrmcyj3AG6Wa_tjM-vPBoxuezv6SaFImURfd3ZdY25JXVe1nyguM5IHqfnFVSb04U55dYC56yhzA3dXx85vSrGCLkKItDPlPcHC57v_LTK-N_CBzcbAcB7Na0Bm_QY0V_K5TKQUmYGf53Vt7ZDAN9LJDgDmJviz6SDwytFbSuYtEd41Oz278jjgy3QPD0SaXYNb5UmjnWfMEqjG11MKV5mE09xa-oqXfjUL0OOD7OQouoSq0CNAZEADPSgEuMeKQMlLgBpF2Y6oRVquDjxjJtJBkU89UJIllReia_Dw56BoVcnNlI9rN6OyFEq6EHchVG5tZgd_-ukKYoarZ2gTHChQyXccy-t87Nc7AtiaDCcqCE-kY9GzbHx7zrpsQM9QUwPyjKMaoeLpIeejfJ4a5H4TzrfNLUSAgPmq1Z1ORryETB0IR0wk4dlvqbP8DJIIxgQDD7H7GPcZKY-2gq5a9Sn8zvAYtWoq73HCTsySxuZfCwYcXxlE4ivQNYUvLlOfcQhmlp-mmEAi0VD0vKsbLmoPcEHh5UonrJ2MJXgm6X6mjx8AzyZEnNDO8Co5OxzpxlRv4l7pg39xcCOpUFJbvCmHSE5nDZpyGY5sN6WaVIGzpQqCYgRX5AaV33eZp5uc7FOF3xlWuEQACIIk2O78mZNW18STioXliPuRt2kuF_5ktAwmIITxFXC4LmQ8dDTpN_hswdI7Zh8lAOrVhyKp-Lh2D0XR0IFHxsI2RMlx1D2c7s7H6tJUmV1OpQg0FRoMYcHN0guOhje1fdoyIAEqCWV1LXdlc3QtMjDWCQ.m3u8 (HTTPSConnectionPool(host='video-weaver.arn04.hls.ttvnw.net', port=443): Read timed out. (read timeout=20.0))
[cli][info] Stream ended
[cli][info] Closing currently open stream...
error: Error when reading from stream: Read timeout, exiting

**Comment by Zibbp** [comment]
This is an issue with [Streamlink](https://streamlink.github.io/) or your network connection. On the same virtual machine that Ganymede runs, can you [install streamlink](https://streamlink.github.io/install.html#id4) and try saving a live stream?
```bash
streamlink https://twitch.tv/CHANNEL best,best -o foo.mp4
```

This should narrow down what the issue is if this also times out.

**Comment by Elandrya2711** [comment]
I have tried various setups, and I am currently running the application on the same VM. The only change I've made is establishing a VPN connection to my home network. Using my home IP for the connection, I haven't encountered any issues. However, when running it on my PC with a VPN connection to my server, I experience the same problems. It's possible that Twitch might not favor the IP range of my hosting provider (Hetzner).

Although I'm not fond of using the VPN solution, it is working for now. If there's anything specific you'd like me to try to narrow down the issue further, I'm willing to do so. From my perspective, we can consider this issue resolved for now. However, better error logging would be a beneficial improvement.
